src/utils/authentication.ts

```diff
@@ -0,0 +1,4 @@
+export async function getAuthenticator() {
+  const {createDefaultMonokleAuthenticator} = await import('@monokle/synchronizer');
+  return createDefaultMonokleAuthenticator();
+}
```

Comment:
What do you think about something like the below, to only create the authenticator once:
```suggestion
export async function setAuthenticator() {
  const {createDefaultMonokleAuthenticator} = await import('@monokle/synchronizer');
  return createDefaultMonokleAuthenticator();
}

let authenticator: Awaited<ReturnType<typeof setAuthenticator>>;

setAuthenticator().then(auth => {
  authenticator = auth;
});

export function getAuthenticator() {
  return authenticator;
}
```
I mostly tried this as an exercise 😅

Reply:
Good suggestion 👍 I started with something similar, but without "hiding" async nature of this, but then decided to move "instance management" to extension custom runtime context we have. But making this sync would help simplify the code (since extension activation function needs to be sync too).

One thing I'm not sure about though, do you think this could result in race condition, where `getAuthenticator()` is called but `authenticator` instance is not initialized yet? This could results in some hard to debug bugs and instabilities 🤔

Reply:
In fact this gave me idea to change the initialization logic and in the meantime I discovered that plugin activate function could be `async` 😅 🤦 Which simplified it even more.